src/routes/registry.ts

```diff
@@ -316,14 +316,16 @@ const $getSubject = memoize(
 		if (subject) {
 			try {
 				// Try to resolve as a device api key first, using the passed in subject
-				const [device] = (await api.resin.get({
+				const device = (await api.resin.get({
 					resource: 'device',
 					passthrough: { req: permissions.root },
+					id: {
+						// uuids are passed as `d_${uuid}`
+						uuid: subject.replace(/^d_/, ''),
```

Comment:
Not against it, but wanted to say that having the natural key in the `id` as well as a `$filter` a bit lower felt a bit disconnected. 
Maybe it's just the part of me which thinks/translates $filter as a WHERE...

Reply:
It's not in the $filter? The only places it should still be in the $filter if it's something like `uuid: { $in: ... }` or `uuid: { $startswith: '...' }`